web/concrete/src/Service/Configuration/HTTP/NginxGenerator.php

```diff
@@ -0,0 +1,119 @@
+<?php
+namespace Concrete\Core\Service\Configuration\HTTP;
+
+use Concrete\Core\Service\Configuration\GeneratorInterface;
+
+class NginxGenerator implements GeneratorInterface
+{
+    /**
+     * @var Application
+     */
+    protected $app;
+
+    /**
+     * @var array
+     */
+    protected $rules;
+
+    /**
+     * @var array
+     */
+    protected $enabledRules;
+
+    /**
+     * Initializes the instance.
+     *
+     * @param Application $app
+     */
+    public function __construct(Application $app)
```

Comment:
The only difference between NginxGenerator and ApacheGenerator is the constructor, maybe we should have an abstract generator or a generator trait for DRY